types/node/http.d.ts

```diff
@@ -130,6 +130,96 @@ declare module "http" {
     class Server extends NetServer {
         constructor(requestListener?: RequestListener);
         constructor(options: ServerOptions, requestListener?: RequestListener);
+
+        // From http.server
+        addListener(event: 'checkContinue', listener: RequestListener): this;
+        addListener(event: 'checkExpectation', listener: RequestListener): this;
+        addListener(event: 'clientError', listener: (err: Error, socket: Socket) => void): this;
+        addListener(event: 'connect', listener: (req: IncomingMessage, socket: Socket, head: Buffer) => void): this;
+        addListener(event: 'request', listener: RequestListener): this;
+        addListener(event: 'upgrade', listener: (req: IncomingMessage, socket: Socket, head: Buffer) => void): this;
+        // From net.server
+        addListener(event: string, listener: (...args: any[]) => void): this;
```

Comment:
I think we can avoid this copy/paste by moving these events to an interface and use declaration merging instead eg.

```ts
// net.d.ts

interface ServerEvents {
        addListener(event: string, listener: (...args: any[]) => void): this;
        addListener(event: "close", listener: () => void): this;
        addListener(event: "connection", listener: (socket: Socket) => void): this;
        addListener(event: "error", listener: (err: Error) => void): this;
        addListener(event: "listening", listener: () => void): this;
   ...
}
interface Server extends ServerEvents {}
```

```ts
// http.d.ts
interface Server extends HttpBase, NetServerEvents {}
```